src/commands/executeFunction/eventGrid/EventGridFileOpenStep.ts

```diff
@@ -45,23 +45,47 @@ export class EventGridFileOpenStep extends AzureWizardExecuteStep<EventGridExecu
             await ext.context.workspaceState.update('didShowEventGridFileOpenMsg', true);
         }
 
+        // Set a listener to track whether the file was modified before the request is sent
+        let modifiedListenerDisposable: vscode.Disposable;
+        void new Promise<void>((resolve, reject) => {
+            modifiedListenerDisposable = vscode.workspace.onDidChangeTextDocument(async (event) => {
```

Comment:
This looks good and seems to work. One thing to consider though:
1. Do we want to fire an event every time a user edits the same trigger?
2. If we do, I'd say it's probably more valuable to keep a running count of how many times the trigger was edited before sent.

Reply:
Actually, it might be better for this to modify a context and then add it to the context of `executeFunctionWithInput` if possible. That way we just have a running count and the event is only tracked once.

Reply:
Spoken offline: disregard this comment. I'd still prefer it gets added to the context of `executeFunctionWithInput` but not a blocker.